React/src/authConfig.ts

```diff
@@ -0,0 +1,18 @@
+export const msalConfig = {
+    auth: {
+        clientId: process.env.REACT_APP_B2C_CLIENT_ID ?? '',
```

Comment:
```suggestion
        clientId: process.env.REACT_APP_B2C_CLIENT_ID',
```

Đây cũng là 1 bad practice mà cần phải ngưng dùng. `??` là để mình set **valid** default value, thường chỉ sử dụng nếu như vế trái của nó là optional.

Trong trường hợp này cái config bên trái là bắt buộc phải có cho application (nếu ko set thì sẽ có feature breaks), nên  mình ko dc set value thành 1 invalid value như vầy. Cái em đang làm ở đây chỉ đơn thuần là để turn off null check của TypeScript.

Trong các trường hợp configuration thì mình sẽ muốn application phải fail càng sớm càng tốt, và tốt nhất là fail khi start up. Khi đó thì nếu mình deploy ở staging và thiếu configuration thì mình sẽ thấy ngay dc vấn đề để fix, thay vì nếu mình hide issue như vầy thì vấn đề sẽ chỉ xuất hiện khi user thực hiện 1 thao tác nhất định (e.g. login).

Trong C# thì mình có thể viết như vầy để đảm bảo là application mình sẽ fail khi required value ko dc đưa vào:
```
id = value ?? throw new ArgumentException(nameof(value));
```
Trong JavaScript / TypeScript thì ko viết vậy dc nên mình sẽ phải check trước khi cái value đó dc dùng và throw exception.